guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -26,511 +25,316 @@
+            inputStreamToFile(pomStream,
+                              pomFile);
 
+            deployPom(gav,
+                      pomFile);
         } finally {
             try {
                 pomFile.delete();
-            } catch ( Exception e ) {
-                log.warn( "Unable to remove temporary file '" + pomFile.getAbsolutePath() + "'" );
+            } catch (Exception e) {
+                log.warn("Unable to remove temporary file '" + pomFile.getAbsolutePath() + "'");
             }
         }
     }
 
-    public void deployParentPom( final GAV gav ) {
-        //Write pom.xml to temporary file for deployment
-        final File pomXMLFile = new File( System.getProperty( "java.io.tmpdir" ),
-                                          toFileName( gav,
-                                                      "pom.xml" ) );
-
+    private void inputStreamToFile(final InputStream inputStream,
+                                   final File file) {
         try {
-
-            String pomXML = generateParentPOM( gav );
-            try {
-                if ( !pomXMLFile.exists() ) {
-                    pomXMLFile.getParentFile().mkdirs();
-                    pomXMLFile.createNewFile();
-                }
-
-                FileOutputStream fos = new FileOutputStream( pomXMLFile );
-                IOUtils.write( pomXML,
-                               fos );
-
-                fos.flush();
-                fos.close();
-            } catch ( IOException e ) {
-                throw new RuntimeException( e );
+            if (!file.exists()) {
+                file.getParentFile().mkdirs();
+                file.createNewFile();
             }
-            //pom.xml Artifact
-            Artifact pomXMLArtifact = new DefaultArtifact( gav.getGroupId(),
-                                                           gav.getArtifactId(),
-                                                           "pom",
-                                                           gav.getVersion() );
-            pomXMLArtifact = pomXMLArtifact.setFile( pomXMLFile );
-
-            try {
-                //Install into local repository
-                final InstallRequest installRequest = new InstallRequest();
-                installRequest
-                        .addArtifact( pomXMLArtifact );
-
-                Aether.getAether().getSystem().install( Aether.getAether().getSession(),
-                                                        installRequest );
-            } catch ( InstallationException e ) {
-                throw new RuntimeException( e );
+            FileOutputStream fos = new FileOutputStream(file);
+
+            final byte[] buf = new byte[BUFFER_SIZE];
+            int byteRead = 0;
+            while ((byteRead = inputStream.read(buf)) != -1) {
+                fos.write(buf,
+                          0,
+                          byteRead);
             }
+            fos.flush();
+            fos.close();
```

Comment:
Although this is probably just code formatting, could you please make sure the `fos` is closed properly in `finally` block?